gleam_stdlib/test/list_test.gleam

```diff
@@ -201,3 +201,47 @@ pub fn zip_test() {
   list:zip([5,6,7], [1,2])
   |> expect:equal(_, [{5,1}, {6,2}])
 }
+
+pub fn intersperse_test() {
+  list:intersperse([1,2,3], 4)
+  |> expect:equal(_, [1,4,2,4,3,4])
```

Comment:
There should not be a 4 at the end of this list, out only goes between elements.

Reply:
@lpil Oh, right!

Reply:
Sorry for not making this clear :)

Reply:
It was clear, I was dumb ;)